common/data_refinery_common/models/models.py

```diff
@@ -560,6 +560,11 @@ def save(self, *args, **kwargs):
         self.last_modified = current_time
         return super(ComputationalResult, self).save(*args, **kwargs)
 
+    def remove_computed_files_from_s3(self):
+        """ Removes all associated computed files from S3. Use this before deleting a computational result. """
```

Comment:
Would it be better to call this in `pre_delete`? If I recall correctly pre_delete doesn't get called for bulk deletions but it does whenever an individual instance is deleted.

Reply:
yes, it's better... here I used the simplest option which was just to manually remove the files. I'm gonna look into `pre_delete`

Reply:
I was reading [some docs](https://simpleisbetterthancomplex.com/tutorial/2016/07/28/how-to-create-django-signals.html), still not sure how to implement the signals. I think we should put this on a different PR and add it for all models that sync to S3. I'm gonna create an issue for it.